app/models/host.rb

```diff
@@ -1939,17 +1939,15 @@ def self.node_types
   end
 
   def self.openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    ems.empty? ? false : Host.where(:ems_id => ems).exists?
+    ManageIQ::Providers::Openstack::InfraManager::Host.count > 0
   end
 
   def self.non_openstack_hosts_exists?
-    ems = ManageIQ::Providers::Openstack::InfraManager.pluck(:id)
-    Host.where.not(:ems_id => ems).exists?
+    (Host.count - ManageIQ::Providers::Openstack::InfraManager::Host.count) > 0
```

Comment:
@kbrock this is the same code you were just pointing me at yesterday... I was going to merge unless you had concerns.  I still think the methods are strange and it seems weird to have these checks.  It feels like we're missing a method to see if these classes support a feature.